### ISSUE TITLE
Unfortunate comment shift

### ISSUE THREAD

**Issue by jdm** [nrc/rustfmt#272]
```diff
     match headers.get::<AccessControlAllowOrigin>() {
-        Some(&AccessControlAllowOrigin::Any) => true, // Not always true, depends on credentials mode
+        Some(&AccessControlAllowOrigin::Any) => true,
+        // Not always true, depends on credentials mode
         Some(&AccessControlAllowOrigin::Value(ref url)) => req.origin.serialize() == *url,
         Some(&AccessControlAllowOrigin::Null) |
        None => false,
```
The result appears unclear to me - I would usually assume a comment is referring to the subsequent line if it stands alone.

**Comment by nrc** [comment]
Yeah, we shouldn't do this - I think we handle this better for function args and we should apply some of the same logic to match arms.

**Comment by marcusklaas** [comment]
`itemize_list` & `write_list`?

**Comment by nrc** [comment]
@marcusklaas I suspect that match arms are complex enough that we can't use `write_list`, but we'll need to share some of its logic. Might be wrong though.

**Comment by scyptnex** [comment]
#347 fixes this (specifically)

**Comment by nrc** [comment]
Fixed by #347
